lib/module.js

```diff
@@ -142,6 +142,8 @@ Module._findPath = function(request, paths) {
 
   // For each path
   for (var i = 0, PL = paths.length; i < PL; i++) {
+    // Don't search further if path doesn't exist
+    if (paths[i] && internalModuleStat(paths[i]) < 1) continue;
```

Comment:
I guess that one needs the `path._makeLong` fix (https://github.com/nodejs/io.js/pull/1991) ?